control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -4468,9 +4465,17 @@ func (r *HostedControlPlaneReconciler) reconcileCloudControllerManager(ctx conte
 		}
 
 		p := openstack.NewOpenStackParams(hcp)
+
```

Comment:
It might be worth handling the `GetCACertFromCredentialsSecret` in the `ReconcileDeployment` as this is specific to openstack and would maintain the pattern of same function signature for all platforms

Reply:
yeah I know how to do it, and I actually did it at some point but reverted, not sure why. I'll fix it very soon.